ui.apps/src/content/jcr_root/apps/contrib/wcm/components/htmlcontainer/v1/htmlcontainer/htmlcontainer.html

```diff
@@ -0,0 +1,23 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2021 Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div data-sly-use.htmlcontainer="com.adobe.cq.wcm.contrib.components.models.HTMLContainer"
+	 data-sly-use.component="com.adobe.cq.wcm.core.components.models.Component"
+     data-sly-use.template="core/wcm/components/commons/v1/templates.html"
+	 data-sly-test.hasContent="${htmlcontainer.includes}"
+     id="${component.id}"
+     class="cmp-htmlcontainer">
+	<sly data-sly-call="${template.placeholder @ isEmpty=!hasContent, classAppend='cq-marker-start'}"></sly>
```

Comment:
The placeholder should be outside of the element having `data-sly-test.hasContent` so that it gets rendered when there is no content. See https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/text/v2/text/text.html#L26 for an example
I also don't understand what this component outputs. Looks like it will not output anything.